File: src/main/java/me/jasper/jasperproject/JasperItem/Product/Weapons/Burst_Bow.java
package me.jasper.jasperproject.JasperItem.Product.Weapons;

import me.jasper.jasperproject.JasperItem.ItemAttributes.Abilities.Burst_Arrow;
import me.jasper.jasperproject.JasperItem.ItemAttributes.ItemType;
import me.jasper.jasperproject.JasperItem.ItemAttributes.Rarity;
import me.jasper.jasperproject.JasperItem.ItemAttributes.Stats;
import me.jasper.jasperproject.JasperItem.JItem;
import me.jasper.jasperproject.JasperItem.Util.Factory;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;

import java.util.List;
import java.util.Map;

public class Burst_Bow extends JItem implements Factory {
    public Burst_Bow(){
        super("Burst Bow", Material.BOW, Rarity.RARE, ItemType.BOW, 1l, "BURST_BOW");
        Map<Stats, Float> stats = this.getStats();
        stats.put(Stats.DAMAGE, 40f);
        stats.put(Stats.ATTACK_SPEED, 10f);
        stats.put(Stats.CRIT_DAMAGE, 40f);
        stats.put(Stats.CRIT_CHANCE, 5f);
        this.getAbilities().add(new Burst_Arrow(25,5));
    }

    @Override
    public JItem create() {
        return this;
    }

    @Override
    protected List<Component> createLore() {
        return List.of();
    }
}

File: src/main/java/me/jasper/jasperproject/JMinecraft/Item/Product/Weapons/Burst_Bow.kt
package me.jasper.jasperproject.JMinecraft.Item.Product.Weapons

import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Abilities.Burst_Arrow
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.ItemType
import me.jasper.jasperproject.JMinecraft.Item.ItemAttributes.Rarity
import me.jasper.jasperproject.JMinecraft.Item.JItem
import me.jasper.jasperproject.JMinecraft.Item.Util.Factory
import me.jasper.jasperproject.JMinecraft.Player.Stats
import net.kyori.adventure.text.Component
import org.bukkit.Material

class Burst_Bow : JItem("Burst Bow", Material.BOW, Rarity.RARE, ItemType.BOW, "BURST_BOW"),
    Factory {
    init {
        val stats = this.stats
        stats[Stats.DAMAGE] = 40f
        stats[Stats.STRENGTH] = 20f
        stats[Stats.ATTACK_SPEED] = 10f
        stats[Stats.CRIT_DAMAGE] = 40f
        stats[Stats.CRIT_CHANCE] = 5f
        abilities.add(Burst_Arrow(25f, 5))
    }

    override fun create(): JItem {
        return this
    }

    override fun createLore(): List<Component> {
        return listOf()
    }
}
